state.py

```diff
@@ -66,10 +68,8 @@ def update_selected(self, pos):
 
     def perform_move(self, src: Position, dst: Position):
         """Perform move from source to destination and update state, if it is source occupiers turn."""
-        player = self.check_whose_turn()
-        if player == self.board.occupying_player(src):
-            previous_piece = self.board.move(src, dst)
-            self.moves.append(((src, dst), previous_piece))
+        previous_piece = self.board.move(src, dst)
+        self.moves.append(((src, dst), previous_piece))
```

Comment:
Until it is possible to toggle rules, it's too convenient to be able to break rules to set up positions to debug